### PR TITLE
gh-63207: Use GetSystemTimePreciseAsFileTime() in time.time()

### PR DESCRIPTION
**Pull request by vstinner**
----
📚 Documentation preview 📚: https://cpython-previews--116822.org.readthedocs.build/

* Issue: gh-63207

### CODE REVIEW

**Comment by vstinner** [comment]
cc @pitrou @eryksun @zooba

**Comment by vstinner** [comment]
Clock resolution measured by the script: https://github.com/python/cpython/pull/116781#issuecomment-1997815440

Before:

```
time():
- efffective resolution: 15 ms
- implementation: GetSystemTimeAsFileTime()
- resolution: 16 ms

perf_counter():
- efffective resolution: 100 ns
- implementation: QueryPerformanceCounter()
- resolution: 100 ns

monotonic():
- efffective resolution: 100 ns
- implementation: QueryPerformanceCounter()
- resolution: 100 ns
```

After:

```
time():
- efffective resolution: 238 ns
- implementation: GetSystemTimePreciseAsFileTime()
- resolution: 100 ns

perf_counter():
- efffective resolution: 100 ns
- implementation: QueryPerformanceCounter()
- resolution: 100 ns

monotonic():
- efffective resolution: 100 ns
- implementation: QueryPerformanceCounter()
- resolution: 100 ns
```

`time.time()` resolution is way better!  238 ns instead of 15.6 ms, it's around 65 500x more accurate!

**Comment by vstinner** [comment]
On a release build (build using `PCbuild\build.bat -e -p x64` command), I confirm that `GetSystemTimePreciseAsFileTime()` ("change") is slower than `GetSystemTimeAsFileTime()` ("ref"):

```
vstinner@WIN C:\victor\python\main>env\Scripts\python -m pyperf compare_to ref.json change.json                                           
Mean +- std dev: [ref] 35.8 ns +- 0.8 ns -> [change] 52.9 ns +- 2.2 ns: 1.48x slower
```

Well, more and more operating have two clocks: one slow but accurate, and one fast but less accurate. Example on Linux:

* CLOCK_REALTIME
* CLOCK_REALTIME_COARSE: A faster but less precise version of CLOCK_REALTIME

Python uses CLOCK_REALTIME for time.time(), the most precise one. So it's consistent to use the most accurate clock for time.time() on Windows as well.

If tomorrow, performance becomes an issue, we can add a new `time.time_coarse()` function which would use CLOCK_REALTIME_COARSE on Linux, and GetSystemTimeAsFileTime() on Windows.

**Comment by pitrou** [comment]
> `time.time()` resolution is way better!  238 ns instead of 15.6 ms, it's around 65 500x more accurate!

Impressive! Finally Windows will be as good as other platforms here.

File: Python/pytime.c [hunk 1]
```diff
@@ -55,6 +55,14 @@
 #endif
 
 
+#ifdef MS_WINDOWS
```
Comment (eryksun) [inline Python/pytime.c:58]: Yes, I think checking for `MS_WINDOWS` here is sufficient. `GetSystemTimePreciseAsFileTime()`, `QueryPerformanceCounter()`, and `QueryPerformanceFrequency()` are available in the system, app, and game partitions. The old use of `GetSystemTimeAdjustment()` actually wasn't supported for app builds. However, that function was used improperly anyway.

**Review by eryksun — APPROVED** [review]

**Pull request merged**